lib/galaxy/config/__init__.py

```diff
@@ -371,7 +370,7 @@ def __init__(self, **kwargs):
         # should probably drop the backward compatiblity to save the path check.
         self.container_image_cache_path = resolve_path(kwargs.get("container_image_cache_path", "container_images"), self.data_dir)
         if not os.path.exists(self.container_image_cache_path):
-            self.container_image_cache_path = self.resolve_path(kwargs.get("container_image_cache_path", "database/container_cache"))
+            self.container_image_cache_path = resolve_path(kwargs.get("container_image_cache_path", "container_cache"), self.data_dir)
```

Comment:
Wouldn't this break setups where in `galaxy.yml` this property is configured to a path relative to the Galaxy root?
```suggestion
            self.container_image_cache_path = self.resolve_path(kwargs.get("container_image_cache_path", os.path.join(self.data_dir, "container_cache")))
```

Reply:
I assumed data_dir was absolute - but I guess not. I based this on some other code - I wonder if they were handled differently somewhere else though.

Reply:
`self.data_dir` should be absolute, but `container_image_cache_path` could be set to a relative path, which should be still be interpreted w.r.t. `self.root`, not `self.data_dir` .